src/game_battler.cpp

```diff
@@ -384,8 +384,8 @@ int Game_Battler::ApplyConditions() {
 	for (int16_t inflicted : GetInflictedStates()) {
 		// States are guaranteed to be valid
 		lcf::rpg::State& state = *lcf::ReaderUtil::GetElement(lcf::Data::states, inflicted);
-		int hp = state.hp_change_val + (int)(std::ceil(GetMaxHp() * state.hp_change_max / 100.0));
-		int sp = state.sp_change_val + (int)(std::ceil(GetMaxHp() * state.sp_change_max / 100.0));
+		int hp = state.hp_change_val + (GetMaxHp() * state.hp_change_max / 100.0);
+		int sp = state.sp_change_val + (GetMaxSp() * state.sp_change_max / 100.0);
```

Comment:
There is no rounding done here by RPG_RT so this change is correct. However, RPG_RT also does integer division. Please change `100.0` to `100` and this will be correct.

Reply:
Did the change and updated this PR.